lib/workers/repository/init/index.ts

```diff
@@ -22,6 +23,11 @@ export async function initRepo(input: RenovateConfig): Promise<RenovateConfig> {
   config.semanticCommits = await detectSemanticCommits(config);
   config = await checkOnboardingBranch(config);
   config = await mergeRenovateConfig(config);
+
+  for (const key of redactedFields) {
```

Comment:
Should it be done within the merge function so that we don't miss nested ones?

Reply:
sounds good, will the merge be done before decrypt?

Reply:
I forget. Alternatively we could traverse the final config object to look for all matching fields. That might be less work than checking every time during merge.

Reply:
We sanitize on log write too
https://github.com/renovatebot/renovate/blob/3c0f77627e4d3c593d1a2043bcf5b0ad296e9bdd/lib/logger/utils.ts#L78-L79

Reply:
removed